### PR TITLE
Refactor index schema change test.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
1 refactor index schema change test case.
2 tiny clean up.

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
@siddontang  PTAL

File: ddl/index_test.go [hunk 1]
```diff
@@ -155,153 +173,436 @@ func testGetIndex(t table.Table, name string) *column.IndexedCol {
...
+
+	switch state {
+	case model.StateNone:
+		s.checkNoneIndex(c, ctx, d, tblInfo, handle, indexCol, row)
+	case model.StateDeleteOnly:
+		s.checkDeleteOnlyIndex(c, ctx, d, tblInfo, handle, indexCol, row, isDropped)
+	case model.StateWriteOnly:
+		s.checkWriteOnlyIndex(c, ctx, d, tblInfo, handle, indexCol, row, isDropped)
+	case model.StateReorganization:
+		// do nothing
```
Comment (siddontang) [inline ddl/index_test.go]: why no reorg check?

**Comment by qiuyesuifeng** [comment]
@siddontang  PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**